pkg/plugin/processor/builtin/renameField.go

```diff
@@ -0,0 +1,149 @@
+
+package builtin
+
+import (
+	"context"
+	"strings"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type renameField struct {
+	mapping map[string]string
+
+	sdk.UnimplementedProcessor
+}
+
+func (p *renameField) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "field.rename",
+		Summary: "rename a group of fields",
+		Description: "rename a group of fields. It is not allowed to rename top-level fields (.Operation, " +
+			".Position, .Key, .Metadata, .Payload.Before, .Payload.After), processor only runs on structured data.",
+		Version: "v1.0",
+		Author:  "Meroxa, Inc.",
+		Parameters: map[string]sdk.Parameter{
+			"mapping": {
+				Default: "",
+				Type:    sdk.ParameterTypeString,
+				Description: "a comma separated list of keys and values, for fields and their new names, " +
+					"ex: .Metadata.key:id,.Payload.After.foo:bar",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					}, {
+						Type: sdk.ValidationTypeExclusion,
+						Value: MetadataReference + "," + PayloadReference + "," + PayloadBeforeReference + "," +
+							PayloadAfterReference + "," + PositionReference + KeyReference + "," + OperationReference,
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *renameField) Configure(_ context.Context, cfg map[string]string) error {
+	list, ok := cfg["mapping"]
+	if !ok || list == "" {
+		return cerrors.Errorf("%w (%q)", ErrRequiredParamMissing, "mapping")
+	}
+
+	result := make(map[string]string)
+	pairs := strings.Split(list, ",")
+	for _, pair := range pairs {
+		parts := strings.Split(pair, ":")
+		if len(parts) != 2 {
+			return cerrors.Errorf("wrong format for the %q param, should be a comma separated list of keys and values,"+
+				"ex: .Metadata.key:id,.Payload.After.foo:bar", "mapping")
+		}
+		key := strings.TrimSpace(parts[0])
+		if key == OperationReference || key == KeyReference || key == PayloadReference || key ==
+			PayloadBeforeReference || key == PayloadAfterReference || key == PositionReference || key == MetadataReference {
+			return cerrors.Errorf("cannot rename one of the top-level fields %q", key)
+		}
+		value := strings.TrimSpace(parts[1])
+		result[key] = value
+	}
+	p.mapping = result
+
+	return nil
+}
+
+func (p *renameField) Open(context.Context) error {
+	return nil
+}
+
+func (p *renameField) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, record := range records {
+		for key, val := range p.mapping {
+			err := p.rename(record, key, val)
+			if err != nil {
+				return append(out, sdk.ErrorRecord{Error: err})
+			}
+		}
+		out = append(out, sdk.SingleRecord(record))
+	}
+	return out
+}
+
+func (p *renameField) rename(record opencdc.Record, key, val string) error {
+	resolver1, err := sdk.NewReferenceResolver(key)
```

Comment:
Similarly to another processor, it would be good to have this initialized once in Configure(), and then used here.

Reply:
this one also could have multiple fields to rename, so I'll go with a slice of resolvers in the `Configure` method